libkbfs/journal_server.go

```diff
@@ -311,55 +313,86 @@ func (j *JournalServer) EnableExistingJournals(
 		return err
 	}
 
-	for _, fi := range fileInfos {
-		name := fi.Name()
-		if !fi.IsDir() {
-			j.log.CDebugf(ctx, "Skipping file %q", name)
-			continue
-		}
-
-		dir := filepath.Join(j.rootPath(), name)
-		uid, key, tlfID, err := readTLFJournalInfoFile(dir)
-		if err != nil {
-			j.log.CDebugf(
-				ctx, "Skipping non-TLF dir %q: %+v", name, err)
-			continue
-		}
-
-		if uid != currentUID {
-			j.log.CDebugf(
-				ctx, "Skipping dir %q due to mismatched UID %s",
-				name, uid)
-			continue
+	eg, groupCtx := errgroup.WithContext(ctx)
+
+	fileCh := make(chan os.FileInfo, len(fileInfos))
+	worker := func() error {
+		for fi := range fileCh {
+			name := fi.Name()
```

Comment:
Cool thx. For future reference you can link to https://github.com/keybase/kbfs/pull/889/files?w=1 (note the w=1) to see the whitespace-agnostic diff (but we have to comment on the regular diff).